app/src/main/java/app/grapheneos/camera/ui/activities/MainActivity.kt

```diff
@@ -1058,16 +1058,35 @@ open class MainActivity : AppCompatActivity(),
     private lateinit var dialog: Dialog
 
     open fun bytesToHex(bytes: ByteArray): String {
-        val hexChars = CharArray(bytes.size)
-        for (i in 0 until bytes.indices.last - 1 step 3) {
-            val v: Int = bytes[i].toInt() and 0xFF
-            hexChars[i] = hexArray[v ushr 4]
-            hexChars[i + 1] = hexArray[v and 0x0F]
-            hexChars[i + 2] = ' '
+        // Add a space between every byte
+        // Double space every 4 bytes (unless end or newline)
+        // Add a newline every 8 bytes (unless end)
```

Comment:
(optional): it might be better to make it clear that the output is formatted as a matrix